internal/jobcontainers/mounts.go

```diff
@@ -56,3 +57,32 @@ func setupMounts(spec *specs.Spec, sandboxVolumePath string) error {
 
 	return nil
 }
+
+// setupMounts sets up all requested mounts present in the OCI runtime spec. They will be mounted from
+// mount.Source to mount.Destination as well as mounted from mount.Source to under the rootfs location
+// for backwards compat with systems that don't have the Bind Filter functionality available.
+func (c *JobContainer) setupMounts(spec *specs.Spec) error {
+	for _, mount := range spec.Mounts {
+		if mount.Destination == "" || mount.Source == "" {
+			return fmt.Errorf("invalid OCI spec - a mount must have both source and a destination: %+v", mount)
+		}
+
+		if isnamedPipePath(mount.Source) {
+			return errors.New("named pipe mounts not supported for job containers - interact with the pipe directly")
+		}
+
+		if err := c.job.ApplyFileBinding(mount.Destination, mount.Source, false); err != nil {
```

Comment:
Does ApplyFileBinding() fail if mount.Destination is a non-empty directory on the host?
I think we discussed about failing with an easy to understand error message if the destination does exist on the host to have deterministic behaviors when interacting with files in the volume mounts.

Reply:
No it doesn't at the moment. Wondering if putting the logic in ApplyFileBinding is the right spot though (but agree we should check if it's an empty dir for this use case)

Reply:
Maybe we just change the third argument to take in a settings struct instead to make this more configurable.

Reply:
We could check right here and return a nice error message.

Reply:
Right, was more wondering if we *could* make `ApplyFileBinding` handle this nicely as we already stat inside that call which if the caller knew the path already existed and was good to go is unnecessary. I think that can be a followup PR, it's an internal API right now anyways. I'll check here

Reply:
Thinking about this more, was the concern originally that there'd be an interleaving of host files and the mounts files in one directory? If so, the behavior here is actually what a normal containers would be, the directory will get shadowed instead of having a merged view. Do you still think we should check this here?

Reply:
@marosset Realized you were out when I messaged this (😆) so friendly ping

Reply:
What are the implications of having a shadowed view of the directory instead of a merged view of the directory?

My main intention here is that I do not ever want the behaviors of workloads running inside a host process container to be dependent on the state of the host machine. Host-process containers have a lot more power to manipulate the host and will also be frequently used in daemonSets so there is a lot of potential for having these containers be re-scheduled.

Reply:
Shadowed would mean the original contents of the directory aren't viewable/or interactable from in the container and would be replaced with whatever the contents of the directory we're binding over it with.

Example:
If you had a directory C:\foo that contained bar.txt and had a host mount from C:\baz to C:\foo, bar.txt at C:\foo isn't visible in the container and is replaced with whatever content is in C:\baz. This should be the same behavior as windows server containers where on next run if you were mounting a host path in, the 'state' would just be if anything changed in the host path itself that's being shared in (in this example C:\baz). 

I think we're in agreeance the general attitude for mounts for hpc is to just use the host path directly and we mostly support mounts to work around the pre-defined k8s mounts (C:\var\run\secrets for example). With that in mind, I don't know if it's worth checking anything

Reply:
Can we add some logging around this then?
I want to give users the tools needed to understand why behaviors for volume mounts with host-process containers may be different than expected.

Reply:
The actual behaviors between hpc and ws containers would be the same here, whatever you're mounting over will have the content masked if there was something there and will be replaced with the contents of the host path, although maybe for hpc there's a higher chance of your container also needing something from there?

Reply:
> although maybe for hpc there's a higher chance of your container also needing something from there?

Yes, i think there is a much higher chance for host process containers to have unpredictable behavior because the hosts file system is accessible without needing to use volume mounts.

Reply:
Added a warning log if the mount destination exists here